etl/steps/data/garden/energy/2024-05-08/energy_mix.py

```diff
@@ -282,30 +282,35 @@ def calculate_primary_energy_annual_change(
         Data after adding annual changes.
 
     """
-    primary_energy = primary_energy.reset_index()
+    primary_energy = tb.reset_index()
 
     # Calculate annual change in each source.
     primary_energy = primary_energy.sort_values(["country", "year"]).reset_index(drop=True)
     for source in ONLY_DIRECT_ENERGY:
         # Create column for source percentage growth as a function of direct primary energy.
-        primary_energy[f"{source} (% growth)"] = primary_energy.groupby("country")[f"{source} (TWh)"].pct_change() * 100
+        primary_energy[f"{source} (% growth)"] = (
+            primary_energy.groupby("country", observed=True)[f"{source} (TWh)"].pct_change(fill_method=None) * 100
```

Comment:
[I see fill_method is deprecated since pandas 2.1](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pct_change.html). Is it necessary to add it?

Reply:
That's what the warning was asking me to do. In fact, I see that "Deprecated since version 2.1: All options of fill_method are deprecated except fill_method=None.". So, the method `fill_method=None` is not deprecated.